Utilizaremos el dataset de “Video Game Sales” que contiene información sobre las ventas de videojuegos en diferentes regiones del mundo. El dataset incluye detalles sobre el nombre del juego, la plataforma, el año de lanzamiento, el género, el editor, y las ventas en diferentes regiones.

### 1. Preparación del Entorno

In [ ]:
# Usamos la biblioteca google para poder usar archivos en nuestro drive.
from google.colab import drive
drive.mount('/content/drive')


Mounted at /content/drive


In [ ]:
## importamos la bibioteca pandas
import pandas as pd

## 2. Cargar los Datos

In [ ]:
## importamos la base de datos desde 'DATA' en formato csv y la guardamos en una variable
path = '/content/drive/MyDrive/DATA/vgsales.csv'
df = pd.read_csv(path)
df.head(10)

Rank                       Name Platform    Year         Genre Publisher  \
0     1                 Wii Sports      Wii  2006.0        Sports  Nintendo   
1     2          Super Mario Bros.      NES  1985.0      Platform  Nintendo   
2     3             Mario Kart Wii      Wii  2008.0        Racing  Nintendo   
3     4          Wii Sports Resort      Wii  2009.0        Sports  Nintendo   
4     5   Pokemon Red/Pokemon Blue       GB  1996.0  Role-Playing  Nintendo   
5     6                     Tetris       GB  1989.0        Puzzle  Nintendo   
6     7      New Super Mario Bros.       DS  2006.0      Platform  Nintendo   
7     8                   Wii Play      Wii  2006.0          Misc  Nintendo   
8     9  New Super Mario Bros. Wii      Wii  2009.0      Platform  Nintendo   
9    10                  Duck Hunt      NES  1984.0       Shooter  Nintendo   

   NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  
0     41.49     29.02      3.77         8.46         82.74  
1     29.08      3.58      6.81         0.77         40.24  
2     15.85     12.88      3.79         3.31         35.82  
3     15.75     11.01      3.28         2.96         33.00  
4     11.27      8.89     10.22         1.00         31.37  
5     23.20      2.26      4.22         0.58         30.26  
6     11.38      9.23      6.50         2.90         30.01  
7     14.03      9.20      2.93         2.85         29.02  
8     14.59      7.06      4.70         2.26         28.62  
9     26.93      0.63      0.28         0.47         28.31

## 3. Exploración Inicial de los Datos

In [ ]:
# Muestra las últimas 5 filas del DataFrame.
df.tail(5)

Rank                                              Name Platform  \
16593  16596                Woody Woodpecker in Crazy Castle 5      GBA   
16594  16597                     Men in Black II: Alien Escape       GC   
16595  16598  SCORE International Baja 1000: The Official Game      PS2   
16596  16599                                        Know How 2       DS   
16597  16600                                  Spirits & Spells      GBA   

         Year     Genre   Publisher  NA_Sales  EU_Sales  JP_Sales  \
16593  2002.0  Platform       Kemco      0.01      0.00       0.0   
16594  2003.0   Shooter  Infogrames      0.01      0.00       0.0   
16595  2008.0    Racing  Activision      0.00      0.00       0.0   
16596  2010.0    Puzzle    7G//AMES      0.00      0.01       0.0   
16597  2003.0  Platform     Wanadoo      0.01      0.00       0.0   

       Other_Sales  Global_Sales  
16593          0.0          0.01  
16594          0.0          0.01  
16595          0.0          0.01  
16596          0.0          0.01  
16597          0.0          0.01

### Utiliza el método info() para obtener información general sobre el DataFrame, incluyendo el número de entradas, nombres de las columnas, tipos de datos y memoria utilizada.

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16598 entries, 0 to 16597
Data columns (total 11 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Rank          16598 non-null  int64  
 1   Name          16598 non-null  object 
 2   Platform      16598 non-null  object 
 3   Year          16327 non-null  float64
 4   Genre         16598 non-null  object 
 5   Publisher     16540 non-null  object 
 6   NA_Sales      16598 non-null  float64
 7   EU_Sales      16598 non-null  float64
 8   JP_Sales      16598 non-null  float64
 9   Other_Sales   16598 non-null  float64
 10  Global_Sales  16598 non-null  float64
dtypes: float64(6), int64(1), object(4)
memory usage: 1.4+ MB


 * Se verifica que existen 16.598 filas.
 * Contiene 11 columnas.
 * Tipos de datos: 1 int64, 4 object y 6 float.
 * Con uso de memoria de 1.4 MB.
 * En Columna 'Year' tiene 271 datos nulos.
 * En Columna 'Publisher' tiene 58 datos nulos.

### Genera estadísticas descriptivas del DataFrame utilizando el método describe().

In [ ]:
df.describe().T.round(2)

count     mean      std      min      25%      50%       75%  \
Rank          16598.0  8300.61  4791.85     1.00  4151.25  8300.50  12449.75   
Year          16327.0  2006.41     5.83  1980.00  2003.00  2007.00   2010.00   
NA_Sales      16598.0     0.26     0.82     0.00     0.00     0.08      0.24   
EU_Sales      16598.0     0.15     0.51     0.00     0.00     0.02      0.11   
JP_Sales      16598.0     0.08     0.31     0.00     0.00     0.00      0.04   
Other_Sales   16598.0     0.05     0.19     0.00     0.00     0.01      0.04   
Global_Sales  16598.0     0.54     1.56     0.01     0.06     0.17      0.47   

                   max  
Rank          16600.00  
Year           2020.00  
NA_Sales         41.49  
EU_Sales         29.02  
JP_Sales         10.22  
Other_Sales      10.57  
Global_Sales     82.74

## A partir de la descripcion de los datos numeros se puede establecer que :
  * la cantidad de registros son 16598.
  * El rango de fechas de los datos esta entre 1980 y 2020.
  * La mayores ventas se registraron en 'NA_Sales' con casi la mitad de
    las ventas Globales de columna 'Global_Sales'.
   

## 4. Inspección de los Datos

In [ ]:
df.columns

Index(['Rank', 'Name', 'Platform', 'Year', 'Genre', 'Publisher', 'NA_Sales',
       'EU_Sales', 'JP_Sales', 'Other_Sales', 'Global_Sales'],
      dtype='object')

In [ ]:
# Inspecciona los tipos de datos de cada columna utilizando el atributo dtypes.
df.dtypes

Rank              int64
Name             object
Platform         object
Year            float64
Genre            object
Publisher        object
NA_Sales        float64
EU_Sales        float64
JP_Sales        float64
Other_Sales     float64
Global_Sales    float64
dtype: object

#### Se verifica que existe 4 columnas de tipo object, 6 columnas de tipo float64 y 1 de tipo int64

In [ ]:
#   Cuenta los valores únicos en la columna 'Genre' utilizando el método value_counts().
df['Genre'].value_counts().sort_values()

Genre
Puzzle           582
Strategy         681
Fighting         848
Simulation       867
Platform         886
Racing          1249
Adventure       1286
Shooter         1310
Role-Playing    1488
Misc            1739
Sports          2346
Action          3316
Name: count, dtype: int64

In [ ]:
#  Muestra todos los valores únicos en la columna Platform utilizando el método unique().
df['Platform'].unique()

array(['Wii', 'NES', 'GB', 'DS', 'X360', 'PS3', 'PS2', 'SNES', 'GBA',
       '3DS', 'PS4', 'N64', 'PS', 'XB', 'PC', '2600', 'PSP', 'XOne', 'GC',
       'WiiU', 'GEN', 'DC', 'PSV', 'SAT', 'SCD', 'WS', 'NG', 'TG16',
       '3DO', 'GG', 'PCFX'], dtype=object)

### 5. Filtrado de Datos

In [ ]:
#  Filtra el DataFrame para mostrar solo las filas donde las ventas en América del Norte (NA_Sales) sean mayores a 1 millón.
filtro = df['NA_Sales'] > 1
df[filtro]

Rank                            Name Platform    Year         Genre  \
0        1                      Wii Sports      Wii  2006.0        Sports   
1        2               Super Mario Bros.      NES  1985.0      Platform   
2        3                  Mario Kart Wii      Wii  2008.0        Racing   
3        4               Wii Sports Resort      Wii  2009.0        Sports   
4        5        Pokemon Red/Pokemon Blue       GB  1996.0  Role-Playing   
...    ...                             ...      ...     ...           ...   
1905  1907                           Joust     2600  1982.0      Platform   
1917  1919                NCAA Football 14     X360  2013.0        Sports   
1953  1955                       NFL Blitz      N64  1998.0        Sports   
1954  1956         NFL Quarterback Club 98      N64  1997.0        Sports   
1957  1959  Diablo II: Lord of Destruction       PC  2001.0  Role-Playing   

                  Publisher  NA_Sales  EU_Sales  JP_Sales  Other_Sales  \
0                  Nintendo     41.49     29.02      3.77         8.46   
1                  Nintendo     29.08      3.58      6.81         0.77   
2                  Nintendo     15.85     12.88      3.79         3.31   
3                  Nintendo     15.75     11.01      3.28         2.96   
4                  Nintendo     11.27      8.89     10.22         1.00   
...                     ...       ...       ...       ...          ...   
1905                  Atari      1.01      0.06      0.00         0.01   
1917        Electronic Arts      1.01      0.00      0.00         0.06   
1953           Midway Games      1.02      0.04      0.00         0.01   
1954  Acclaim Entertainment      1.01      0.05      0.00         0.01   
1957          Vivendi Games      1.03      0.02      0.00         0.00   

      Global_Sales  
0            82.74  
1            40.24  
2            35.82  
3            33.00  
4            31.37  
...            ...  
1905          1.08  
1917          1.07  
1953          1.06  
1954          1.06  
1957          1.06  

[895 rows x 11 columns]

 * Se supone que los valores son en millones

In [ ]:
#  Filtra el DataFrame para mostrar solo las filas donde las ventas en Japón (JP_Sales) sean menores a 0.1 millón.
filtro_2 = df['JP_Sales'] < 0.1
df[filtro_2]



Rank                                              Name Platform  \
23        24                                Grand Theft Auto V     X360   
35        36                        Call of Duty: Black Ops II     X360   
36        37                    Call of Duty: Modern Warfare 2     X360   
60        61                                      Just Dance 3      Wii   
61        62                              Call of Duty: Ghosts     X360   
...      ...                                               ...      ...   
16593  16596                Woody Woodpecker in Crazy Castle 5      GBA   
16594  16597                     Men in Black II: Alien Escape       GC   
16595  16598  SCORE International Baja 1000: The Official Game      PS2   
16596  16599                                        Know How 2       DS   
16597  16600                                  Spirits & Spells      GBA   

         Year     Genre             Publisher  NA_Sales  EU_Sales  JP_Sales  \
23     2013.0    Action  Take-Two Interactive      9.63      5.31      0.06   
35     2012.0   Shooter            Activision      8.25      4.30      0.07   
36     2009.0   Shooter            Activision      8.52      3.63      0.08   
60     2011.0      Misc               Ubisoft      6.05      3.15      0.00   
61     2013.0   Shooter            Activision      6.72      2.63      0.04   
...       ...       ...                   ...       ...       ...       ...   
16593  2002.0  Platform                 Kemco      0.01      0.00      0.00   
16594  2003.0   Shooter            Infogrames      0.01      0.00      0.00   
16595  2008.0    Racing            Activision      0.00      0.00      0.00   
16596  2010.0    Puzzle              7G//AMES      0.00      0.01      0.00   
16597  2003.0  Platform               Wanadoo      0.01      0.00      0.00   

       Other_Sales  Global_Sales  
23            1.38         16.38  
35            1.12         13.73  
36            1.29         13.51  
60            1.07         10.26  
61            0.82         10.21  
...            ...           ...  
16593         0.00          0.01  
16594         0.00          0.01  
16595         0.00          0.01  
16596         0.00          0.01  
16597         0.00          0.01  

[14021 rows x 11 columns]

* Analizando el resultado de filas, se verifica que en la Columna 'JP_Sales', hay 14021 filas con sus ventas menores a 0.1 millon.
* ademas se verifica que hay valores en cero

In [ ]:
#  Utilizando el método query(), filtra el DataFrame para mostrar las filas donde el género sea Action y las ventas globales (Global_Sales) sean mayores a 2 millones.

df.query("Genre == 'Action' & Global_Sales > 2")


Rank                                               Name Platform    Year  \
16     17                                 Grand Theft Auto V      PS3  2013.0   
17     18                      Grand Theft Auto: San Andreas      PS2  2004.0   
23     24                                 Grand Theft Auto V     X360  2013.0   
24     25                        Grand Theft Auto: Vice City      PS2  2002.0   
38     39                               Grand Theft Auto III      PS2  2001.0   
..    ...                                                ...      ...     ...   
815   817             Tom Clancy's Splinter Cell: Conviction     X360  2010.0   
817   819                                    The Incredibles      GBA  2004.0   
833   835  Peter Jackson's King Kong: The Official Game o...      PS2  2005.0   
835   837                            Need for Speed: The Run      PS3  2011.0   
840   842                                         Dead Space      PS3  2008.0   

      Genre             Publisher  NA_Sales  EU_Sales  JP_Sales  Other_Sales  \
16   Action  Take-Two Interactive      7.01      9.27      0.97         4.14   
17   Action  Take-Two Interactive      9.43      0.40      0.41        10.57   
23   Action  Take-Two Interactive      9.63      5.31      0.06         1.38   
24   Action  Take-Two Interactive      8.41      5.49      0.47         1.78   
38   Action  Take-Two Interactive      6.99      4.51      0.30         1.30   
..      ...                   ...       ...       ...       ...          ...   
815  Action               Ubisoft      1.20      0.63      0.04         0.19   
817  Action                   THQ      1.15      0.77      0.04         0.10   
833  Action               Ubisoft      0.71      1.02      0.00         0.31   
835  Action       Electronic Arts      0.58      1.06      0.03         0.37   
840  Action       Electronic Arts      1.05      0.67      0.00         0.30   

     Global_Sales  
16          21.40  
17          20.81  
23          16.38  
24          16.15  
38          13.10  
..            ...  
815          2.06  
817          2.06  
833          2.04  
835          2.04  
840          2.02  

[175 rows x 11 columns]

* Se supone que la columna 'Global Sales' esta en millones
* se verifica 175 filas del filtro son mayores a 2 millones


### 6. Slicing de Datos

In [ ]:
# Selecciona y muestra solo las columnas Name y Global_Sales del DataFrame.
df[["Name", "Global_Sales"]]

Name  Global_Sales
0                                            Wii Sports         82.74
1                                     Super Mario Bros.         40.24
2                                        Mario Kart Wii         35.82
3                                     Wii Sports Resort         33.00
4                              Pokemon Red/Pokemon Blue         31.37
...                                                 ...           ...
16593                Woody Woodpecker in Crazy Castle 5          0.01
16594                     Men in Black II: Alien Escape          0.01
16595  SCORE International Baja 1000: The Official Game          0.01
16596                                        Know How 2          0.01
16597                                  Spirits & Spells          0.01

[16598 rows x 2 columns]

In [ ]:
# Utilizando loc[], selecciona y muestra las filas de la 5 a la 10 (inclusive) y las columnas Name y Genre.
res_loc = df.loc[5:10, ["Name", "Genre"]]
print(res_loc)

                         Name       Genre
5                      Tetris      Puzzle
6       New Super Mario Bros.    Platform
7                    Wii Play        Misc
8   New Super Mario Bros. Wii    Platform
9                   Duck Hunt     Shooter
10                 Nintendogs  Simulation


In [ ]:
# Utilizando iloc[], selecciona y muestra las primeras 5 filas y las primeras 3 columnas del DataFrame.
res_iloc = df.iloc[0:5, 0:3]
print(res_iloc)

   Rank                      Name Platform
0     1                Wii Sports      Wii
1     2         Super Mario Bros.      NES
2     3            Mario Kart Wii      Wii
3     4         Wii Sports Resort      Wii
4     5  Pokemon Red/Pokemon Blue       GB
